test/common/http/conn_manager_utility_test.cc

```diff
@@ -128,7 +129,7 @@ class ConnectionManagerUtilityTest : public testing::Test {
                                                        random_, local_info_)
             ->asString();
     ConnectionManagerUtility::mutateTracingRequestHeader(headers, runtime_, config_, &route_);
-    ret.internal_ = headers.EnvoyInternalRequest() != nullptr;
+    ret.internal_ = HeaderUtility::isEnvoyInternalRequest(headers);
```

Comment:
Is there a test that covers the actual regression being fixed?

Reply:
Probably not. This is test.cc. I guess we don't need another test for test.cc :)
Here is the no brain fix.

I add a test case in router/config_impl_test.cc to cover the core change.